### PR TITLE
Improve release process

### PR DESCRIPTION
**Pull request by jacobdotcosta**
* Add release label to the release PR
  *  this will allow filtering the release PR from the changelog
  * requires the `release` label to actually be created on the repository
* Skip the release workflow if the release PR is closed without being merged
* Generate change log and add it's text to the release body

### CODE REVIEW

File: .github/workflows/tag-and-create-release.yaml [hunk 1]
```diff
@@ -35,6 +35,34 @@ jobs:
           git tag -a "$release_version" -m "Version $release_version"
           git push --follow-tags
 
+      - name: Build Changelog
+        uses: mikepenz/release-changelog-builder-action@v4
+        id: generate_changelog
+        with:
+          configurationJson: |
```
Comment (cmoulliard) [inline .github/workflows/tag-and-create-release.yaml:42]: Can we adapt the configuration to also : 
- include the nodejs packages released. See example: https://github.com/q-shift/backstage-plugins/releases/tag/v0.1.28
- use a `changeset` file to be merged within the release note. Why ? Listing the title of the issues within the release note is sometimes not really user friendly vs using such a wording : 
  - Include new field QuarkusVersionListField to list the quarkus versions available in a template. https://github.com/q-shift/backstage-plugins/pull/61
  - Add devApp to the Quarkus Console project https://github.com/q-shift/backstage-plugins/pull/52
  - Refactor the Quarkus Console and Ported the tab bar from the OpenShift Console plugin https://github.com/q-shift/backstage-plugins/pull/57
  **FYI**: Backstage uses the tool `changeset` - https://github.com/changesets/changesets/blob/main/docs/common-questions.md#changesets-are-automatically-generated to enhance the release note: Example: https://github.com/backstage/backstage/releases/tag/v1.23.0

Comment (jacobdotcosta) [reply .github/workflows/tag-and-create-release.yaml:42]: To clarify, the nodejs packages is a static list of the 3 (for now) plugins that are included on the project which are `plugin-quarkus`, `plugin-quarkus-console` and `plugin-quarkus-backend`, correct?

Comment (cmoulliard) [reply .github/workflows/tag-and-create-release.yaml:42]: Yes. I created the list manually but link could be generated as you know the version like the package name from the 3 package.json files

Comment (jacobdotcosta) [reply .github/workflows/tag-and-create-release.yaml:42]: ✅ **DONE**: include the nodejs packages released

**Comment by cmoulliard** [comment]
What you did here looks good to me. 

We can see if we should customize the content of the change(s) using a changeset file in a next PR

@jacobdotcosta

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**